doc: nRF5340 audio fixes (#19135)
nrfconnect/sdk-nrf: applications/nrf5340_audio/doc/building.rst

Building and programming using script
*************************************

The suggested method for building each of the applications and programming it to the development kit is running the :file:`buildprog.py` Python script.
The script automates the process of selecting :ref:`configuration files <nrf53_audio_app_configuration_files>` and building different applications.
This eases the process of building and programming images for multiple development kits.

The script is located in the :file:`applications/nrf5340_audio/tools/buildprog` directory.

Preparing the JSON file
=======================

The script depends on the settings defined in the :file:`nrf5340_audio_dk_devices.json` file.
Before using the script, make sure to update this file with the following information for each development kit you want to use.
This is how the file looks by default:

.. literalinclude:: ../tools/buildprog/nrf5340_audio_dk_devices.json
    :language: json

When preparing the JSON file, update the following fields:

* ``nrf5340_audio_dk_snr`` -- This field lists the SEGGER serial number.

>   You can check this five-digit number on the sticker on the nRF5340 Audio development kit.

Reviewer: The json-file needs the entire 10-digit number to flash. There is a 12-digit number when running nrfutil device list, so if the first 2 zeros are removed we have the correct 10-digit number

>   You can check this five- or ten-digit number on the sticker on the nRF5340 Audio development kit.

Reviewer: It has to be ten-digit. I tested with five, doesn't work

Running the script
==================

> The script can handle building and programming at the same time.

Reviewer 2: ```suggestion
The script handles building and parallel programming of multiple kits.
```

       | **Note:** For FOTA DFU, you must use :ref:`nrf53_audio_app_building_standard`.
   * - Device type (``-d``)
     - Specifies the device type.
     - ``headset``, ``gateway``, ``both``
     - :ref:`nrf53_audio_app_overview_gateway_headsets`

For example, the following command builds headset and gateway applications using the script for the application core with the ``debug`` application version:

.. code-block:: console

   python buildprog.py -c app -b debug -d both

The command can be run from any location, as long as the correct path to :file:`buildprog.py` is given.

The build files are saved in separate subdirectories in the :file:`applications/nrf5340_audio/build` directory.
The script creates a directory for each application version and device type combination.
For example, when running the command above, the script creates the :file:`dev_gateway/build_debug` and :file:`dev_headset/build_debug` directories.

Script parameters for programming
---------------------------------

The script can program the build files as part of the same `python buildprog.py` command used for building.
Use one of the following programming parameters:

> * Programming (``-p`` parameter) -- If you run the building script with this parameter, you can program one or both of the cores after building the files.

Reviewer 2: ```suggestion
* Programming (``-p`` parameter) -- If you run the buildprog script with this parameter, you can program one or both of the cores after building the files.
```

When using the default CIS configuration, if you want to use two headset devices, you must also populate the UICR with the desired channel for each headset.
Use the following commands, depending on which headset you want to populate:

* Left headset (``--val 0``):

   .. code-block:: console

      nrfjprog --memwr 0x00FF80F4 --val 0

* Right headset (``--val 1``):

   .. code-block:: console

      nrfjprog --memwr 0x00FF80F4 --val 1

Select the correct board when prompted with the popup.
Alternatively, you can add the ``--snr`` parameter followed by the SEGGER serial number of the correct board at the end of the ``nrfjprog`` command.

> You can check the serial numbers of the connected devices with the ``nrfutil device list`` command.

Author: I've replaced nrfjprog for this reason in this PR, but I will mention both.
